src/tiledb/cloud/client.py

```diff
@@ -83,10 +84,20 @@ def login(
     :param no_session: don't create a session token on login,
         store instead username/password
     :param threads: number of threads to enable for concurrent requests
+    :param region: the region to run code in
     :return:
     """
     if host is None:
         host = config.default_host
+    if region is not None:
+        # If a region is provided, modify the host.
+        orig_url = urllib3.util.parse_url(host)
+        region_url = urllib3.util.Url(
+            scheme=orig_url.scheme,
+            # TODO: Validate region.
+            host=region + ".aws." + orig_url.host,
```

Comment:
Should we make the region be `<region>.<cloud_provider>` so its future proofed?

Reply:
Yes, good point. I turned the region into an enum, which also helps with validating the given region.

Reply:
We don’t want to make this AWS-specific; since this is just going to replace the `host` parameter in the endpoint, would it make sense to have a set of constants that the user can pass in?

```python

AWS_US_EAST_1 = "https://us-east-1.aws.api.tiledb.com"
# etc.

# Or namespace it with classes:
class AWS:
  US_EAST_1 = "https://us-east-1.aws.api.tiledb.com"

class OtherCloud:
  SOME_REGION: "https://some-region.other.api.tiledb.com"
```

Reply:
My thinking with the enum was that we simply add another one as we support new cloud providers. That said, if y'all prefer constants, then that's fine with me.

Also, in de2950f, I revised the code to expose a separate function instead, `tiledb.cloud.login_to_region()`. This feels more natural to use.